### ISSUE TITLE
Error Processor takes over all application, not just Rest Module

### ISSUE THREAD

**Issue by aramonc** [vasrap/ZF2-Restful-Module-Skeleton#8]
Hey guys, is there a different listener that I could attach the error processor to so that this can be added as a Module of an app instead of the whole application?

As it is right now the error processor gets dispatched regardless of which module the error occurred in.

**Comment by vasrap** [comment]
Thanks for reporting this. It sounds like a bug. I am gonna work on fix and update this issue when done.

**Comment by aramonc** [comment]
Awesome, thank you. Is there anything I can provide to help replicate?

**Comment by vasrap** [comment]
Nah, I think what is happening is clear. But I will let you know if it becomes unclear :)

**Comment by vasrap** [comment]
@aramonc I have a feeling this is not a bug. Try setting the priority for the "errorProcess" method to something lower (lower than your other error listeners, ie -999). To do that go to Module.php and change method "onBootstrap"; you will see the listener attachment code somewhere in the end of that method.

Let me know.

**Comment by jhuet** [comment]
@scaraveos, on my side i tried to change the priority of the listener calling `errorProcess` to -999 but i still get thrown exceptions showing up as json :(

**Comment by vasrap** [comment]
@jhuet do you have anything else listening to the same event?

**Comment by aramonc** [comment]
@scaraveos I think this is working as you described. I need to run a couple more tests to be sure, but I think this can be closed & just added to the documentation.

**Comment by vasrap** [comment]
Thanks for confirming @aramonc, I will add some more comments on that file.

**Comment by chris--** [comment]
@scaraveos I'm also having problems with the errorProcess taking over despite having set the priority to -999. I'm just getting into ZF2, so my solution is just a hack: 
add 

```
public function errorProcess(MvcEvent $e)
{
    $routeMatch = $e->getRouteMatch();
    if (!$routeMatch) {
        return null;
    }
....
```

I am using the ZF2 skeleton app and changed the `route` in `module/Main/config/module.config.php` to `'/rest/:controller[.:formatter][/:id]'`. Thanks for you work!

**Comment by vasrap** [comment]
@chris-- Do you have anything else listening to the "MvcEvent::EVENT_DISPATCH_ERROR" event? If not then "errorProcess" will always handle that event.

**Comment by aramonc** [comment]
@scaraveos that makes sense. Is there any way to have it do the default other than what @chris-- did?

**Comment by chris--** [comment]
@scaraveos I dont really understand how the error delegation works in the [zf skeleton](https://github.com/zendframework/ZendSkeletonApplication) but I dont think there is anything listening on that event.

**Comment by vasrap** [comment]
@aramonc if I understood you correctly; essentially what @chris-- did is handled at line 129 in Module.php in a different way though.

**Comment by trainerbill** [comment]
I tried all of the work arounds listed in this thread and nothing worked for me.  I decided to check the headers for Accept text/html.  If it finds that header it will return null out of the errorProcess function and display HTML view model.  Here is the commit https://github.com/UNOIS/RestfulApiModule/commit/8bbcd4c54a28492a60b92d3f6efb4eb674de3a91
